### ISSUE TITLE
React components are not detected when using ES6 classes

### ISSUE THREAD

**Issue by steida** [yannickcr/eslint-plugin-react#24]
It seems rules are ignored with such pretty valid code

```es6
import DocumentTitle from 'react-document-title'
import PureComponent from '../../lib/purecomponent'
import {msg} from '../intl/store'

export default class About extends PureComponent {

  render() {
    return (
      <DocumentTitle title={msg('titles.about')}>
        <div className="about-page">
          <div>{this.props.fok}</div>
          <p>
            {msg('about.p')}
          </p>
        </div>
      </DocumentTitle>
    )
  }

}
```

**Comment by yannickcr** [comment]
Hi,

Can you be more specific about the rules that are ignored and your `.eslintrc` configuration ?

**Comment by steida** [comment]
https://gist.github.com/steida/8882dc774ccf7b8ce611

I checked only prop-types and self-closing-comp. It worked well with createClass but not in my example.
Btw, you can check it on https://github.com/steida/este

**Comment by yannickcr** [comment]
We are using React.createClass in `prop-types`, `display-name` and `no-multi-comp` to detect React component and to apply the rules on the object.

Obviously we need to figure out a new way to detect React components when using ES6 classes.

**Comment by steida** [comment]
With React subclassing the only possible detection is jsx syntax only. There is no other shape you can use to detect React class I suppose.

**Comment by framerate** [comment]
I may be off base, but is this related to my issue?:

```
class RalphMacchio extends React.Component {
 ...blah blah
}
```

and...

```
React.render(<RalphMacchio />, document.getElementById('rm'));
```

Gives me:

```
src/app.jsx
  41:14  error  'RalphMacchio' is not defined  react/jsx-no-undef
  41:14  error  'RalphMacchio' is not defined  no-undef
```

Forgive me (and point me in the right direction) if this is unrelated. I'm trying to switch to ESLint to support my new ES6 projects. Here's my WIP .eslintrc:

```
{
    "parser": "babel-eslint",
    "env": {
        "node": true,
        "es6": true
    },

    "plugins": [
        "react"
    ],

    "settings": {
        "ecmascript": 6
    },

    "ecmaFeatures": {
        "jsx": true
    },

    "rules": {
        "no-undef": 2,
        "react/jsx-no-undef": 2,
        "no-unused-vars": 2,
        "react/jsx-uses-vars": 1,
        "no-console": 1,
        "eqeqeq": [2, "smart"],
        "quotes": [2, "single"]
    },

    "globals": {
        "document": true,
        "window": true,
        "_": true
    }
}
```

**Comment by yannickcr** [comment]
@framerate 2 things here:

It seems related to #19 for the `no-undef` error and it should be fixed in next ESLint release.

But the `react/jsx-no-undef` error seems to be a different one and I only reproduce it with babel-eslint as a parser. Can you open an issue for this case ?

**Comment by framerate** [comment]
Sure thing!

**Comment by framerate** [comment]
posted #30. 

My goal here is to write ES6 + React 0.13 and have my code pass ESLint. Other than IGNORING undef, do I have another option yet or am I one the right track?
